src/main/java/org/fogbowcloud/arrebol/ArrebolController.java

```diff
@@ -1,59 +1,78 @@
-import org.fogbowcloud.arrebol.models.specification.Specification;
 import org.fogbowcloud.arrebol.models.task.Task;
 import org.fogbowcloud.arrebol.models.task.TaskState;
 import org.fogbowcloud.arrebol.queue.TaskQueue;
 import org.fogbowcloud.arrebol.repositories.JobRepository;
-import org.fogbowcloud.arrebol.resource.MatchAnyWorker;
 import org.fogbowcloud.arrebol.resource.StaticPool;
 import org.fogbowcloud.arrebol.resource.WorkerPool;
 import org.fogbowcloud.arrebol.scheduler.DefaultScheduler;
 import org.fogbowcloud.arrebol.scheduler.FifoSchedulerPolicy;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
-import java.io.File;
-import java.io.FileInputStream;
-import java.io.IOException;
-import java.io.InputStream;
-import java.util.*;
-
 @Component
 public class ArrebolController {
 
+    private static final int COMMIT_PERIOD_MILLIS = 1000 * 20;
+    private static final int UPDATE_PERIOD_MILLIS = 1000 * 10;
+    private static final int FAIL_EXIT_CODE = 1;
     private final Logger LOGGER = Logger.getLogger(ArrebolController.class);
-
-    private final Properties properties;
     private final DefaultScheduler scheduler;
     private final Map<String, Job> jobPool;
     private final TaskQueue queue;
-
     private final Timer jobDatabaseCommitter;
-
-    private static final int COMMIT_PERIOD_MILLIS = 1000 * 20;//20 seconds
-
+    private final Timer jobStateMonitor;
+    private Configuration configuration;
+    private WorkerCreator workerCreator;
     @Autowired
     private JobRepository jobRepository;
 
-    public ArrebolController(Properties properties) {
-        this.properties = properties;
-
-        String path = Thread.currentThread().getContextClassLoader().getResource("").getPath();
-        Properties arrebolProperties = new Properties();
-
-        try (InputStream input = new FileInputStream(path + File.separator + "arrebol.conf")) {
-            arrebolProperties.load(input);
-        } catch (IOException ex) {
-            LOGGER.error("Error on loading properties file path=" + path, ex);
-            System.exit(1);
+    public ArrebolController() {
+
+        String path = Objects.requireNonNull(Thread.currentThread().getContextClassLoader()
+            .getResource("")).getPath();
+        try {
+            Gson gson = new Gson();
+            BufferedReader bufferedReader = new BufferedReader(
+                new FileReader(path + File.separator + "arrebol.json"));
+            this.configuration = gson.fromJson(bufferedReader, Configuration.class);
+
+            certifyConfigurationProperties();
+
+            DockerVariable.DEFAULT_IMAGE = this.configuration.getImageId();
+            setWorkerCreator(this.configuration.getPoolType());
+        } catch (IOException e) {
+            LOGGER.error("Error on loading properties file path=" + path, e);
+            System.exit(FAIL_EXIT_CODE);
+        } catch (Exception e) {
```

Comment:
Shouldn't be a Throwable catch (I guess you guys mean to fail properly in case of any exception)

Reply:
I also wonder if we should make the try to cover all the constructor

Reply:
I've changed IOException to FileNotFoundException to catch specifically the case of not finding the configuration file.

Reply:
Done.